code/obj/artifacts/artifact_items/dimensional_key.dm

```diff
@@ -0,0 +1,388 @@
+		fissure_area.fissure_region = artkey.fissure_region
+
+		mirrored_physical_zone_created = TRUE
+
+/****** Supporting items/atoms/etc. *******/
+
+ABSTRACT_TYPE(/obj/art_fissure_objs)
+/obj/art_fissure_objs
+	name = ""
+	desc = ""
+	anchored = ANCHORED_ALWAYS
+
+ABSTRACT_TYPE(/obj/art_fissure_objs/cross_dummy)
+/obj/art_fissure_objs/cross_dummy
+	invisibility = INVIS_ALWAYS
+	var/turf/exit_turf
+	var/required_dir
+
+	New(newLoc, turf/exit)
+		..()
+		src.exit_turf = exit
+
+	disposing()
+		src.exit_turf = null
+		..()
+
+	Crossed(atom/movable/AM)
+		if (istype(AM, /obj/projectile))
+			var/obj/projectile/P = AM
+			var/obj/projectile/new_proj = initialize_projectile(src.exit_turf, P.proj_data, P.xo, P.yo, P.shooter)
+			new_proj.travelled = P.travelled
+			new_proj.launch()
+			P.die()
+		else if (AM.dir == src.required_dir && !istype(AM, /obj/art_fissure_objs/door) && !istype(AM, /obj/art_fissure_objs/cross_dummy))
+			AM.set_loc(src.exit_turf)
+		else
+			return ..()
+
+	north
+		required_dir = NORTH
+
+	east
+		required_dir = EAST
+
+	south
+		required_dir = SOUTH
+
+	west
+		required_dir = WEST
+
+/obj/art_fissure_objs/mirror_update_dummy
+	invisibility = INVIS_ALWAYS
+	var/turf/station_ref
+	var/entrance_loc
+
+	New(newLoc, turf/station_ref, entrance_loc)
+		..()
+		src.station_ref = station_ref
+		src.entrance_loc = entrance_loc
+
+	disposing()
+		src.station_ref = null
+		..()
+
+	Crossed(atom/movable/AM)
+		if (isliving(AM) && !isintangible(AM))
+			var/area/artifact_fissure/fissure_area = get_area(src)
+			fissure_area.update_visual_mirrors(src.station_ref, src.entrance_loc)
+			return ..()
+		return ..()
+
+/obj/art_fissure_objs/door
+	name = "mysterious wooden door"
+	desc = "A wooden door, but it emanates some aura. Something's not right about it."
+	icon = 'icons/obj/doors/door_wood.dmi'
+	icon_state = "door1"
+	opacity = TRUE
+	density = TRUE
+	/// open or closed
+	var/open = FALSE
+	/// whether this is a "fake" door or not
+	var/can_be_opened = TRUE
+	/// if a door as a part of the fissure, whether the corresponding outer entrance has been created or not
+	var/outer_entrance_spawned = FALSE
+	/// associated door in/outside the fissure
+	var/obj/art_fissure_objs/door/linked_door = null
+
+	disposing()
+		src.linked_door = null
+		..()
+
+	attack_hand(mob/user)
+		..()
+		if (!src.can_be_opened)
+			src.deny_open()
+			boutput(user, SPAN_NOTICE("[src] is locked. Perhaps a key will open it?"))
+			return
+		if (src.open)
+			src.close()
+		else if (!src.open && istype(get_area(user), /area/artifact_fissure))
+			if (!src.outer_entrance_spawned)
+				src.deny_open()
+				boutput(user, SPAN_NOTICE("[src] doesn't seemed to be locked, but won't open either... strange."))
+			else
+				src.open()
+		else
+			src.deny_open()
+			boutput(user, SPAN_NOTICE("[src] is locked. Perhaps a key will open it?"))
+
+	attackby(obj/item/I, mob/user)
+		if (!istype(I, /obj/item/artifact/dimensional_key) || !I.artifact.activated)
+			return ..()
```

Comment:
intentional that any key art can be used, and not just the one associated with the door?

Reply:
Intentional yeah